File: L1J/src/Stove.java

public class Stove {
	private Pan pan;
    private boolean state ;
    public boolean GetState(){
    	return state;
    }
    public void SetState(boolean s){
    	state=s;
    }
    
    
    
  //  public Pan Pan { set { pan = value; } get { return pan; } }
public Pan GetPan(){
	return pan;
}
public void SetPan(Pan p){
	pan=p;
}

    
    
    
    
    public void Cooking()
    {
        if (GetState()==true) // если плита включена 
        {
            while (!pan.IsReady()) // пока не будет готово 
            {
                pan.Cook();
            }
        }
    }
}

File: Прогр1(1)/Прогр1(1)/Stove.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Прогр1_1_
{
    class Stove
    {
        private Pan pan;
        private bool state;
        public bool State { set { state = value; } get { return state; } }
        public Pan Pan { set { pan = value; } get { return pan; } }

        public void Cooking()
        {
            if (State)
            {
                while (!pan.IsReady())
                {
                    pan.Cook();
                }
            }
        }
    }
}
